testing-cci-datasets/generate_summary.py

```diff
@@ -68,123 +71,49 @@ def get_list_of_ecvs(data_sets):
                 else:
-                    open_bbox_fail += 1
-
-                if 'yes' in dataset['cache(4)']:
-                    cache_success += 1
-                else:
-                    cache_fail += 1
-
-                if 'yes' in dataset['map(5)']:
-                    visualize_success += 1
-                else:
-                    visualize_fail += 1
-            else:
-                not_supported += 1
+                    column_failed = f'{column}_failed'
+                    summary_row_new[column_failed] += 1
         total_number_of_datasets = len(data_sets)
 
-    supported_percentage = 100 * supported / total_number_of_datasets
-    try:
-        open_success_percentage = \
-            100 * open_success / (total_number_of_datasets - not_supported)
-        open_temp_success_percentage = \
-            100 * open_temp_success / (total_number_of_datasets - not_supported)
-        open_bbox_success_percentage = \
-            100 * open_bbox_success / (total_number_of_datasets - not_supported)
-        visualize_success_percentage = \
-            100 * visualize_success / (total_number_of_datasets - not_supported)
-        cache_success_percentage = \
-            100 * cache_success / (total_number_of_datasets - not_supported)
-    except ZeroDivisionError:
-        open_success_percentage = 0.0
-        open_temp_success_percentage = 0.0
-        open_bbox_success_percentage = 0.0
-        visualize_success_percentage = 0.0
-        cache_success_percentage = 0.0
-    summary_row_new = {'ecv': ecv,
-                       'supported': supported,
-                       'open_success': open_success,
-                       'open_fail': open_fail,
-                       'open_temp_success': open_temp_success,
-                       'open_temp_fail': open_temp_fail,
-                       'open_bbox_success': open_bbox_success,
-                       'open_bbox_fail': open_bbox_fail,
-                       'cache_success': cache_success,
-                       'cache_fail': cache_fail,
-                       'visualize_success': visualize_success,
-                       'visualize_fail': visualize_fail,
-                       'supported_percentage': supported_percentage,
-                       'open_success_percentage': open_success_percentage,
-                       'open_temp_success_percentage': open_temp_success_percentage,
-                       'open_bbox_success_percentage': open_bbox_success_percentage,
-                       'visualize_success_percentage': visualize_success_percentage,
-                       'cache_success_percentage': cache_success_percentage,
-                       'total_number_of_datasets': total_number_of_datasets
-                       }
+    for i in summary_columns:
+        try:
+            summary_row_new[f'{i}_percentage'] = 100 * summary_row_new[i] / (
+                    total_number_of_datasets - summary_row_new[
```

Comment:
I know you just adapted the formula that we already had, but shouldn't it be

summary_row_new[f'{i}_percentage'] = 100 * summary_row_new[i] / (total_number_of_datasets)

?

Reply:
No, the idea was to know, how many of the supported datasets are successful with e.g. opening. Because here we don't want to include the datasets which we exclude from support.